raiden/tests/integration/long_running/flaky/test_token_networks.py

```diff
@@ -5,11 +5,54 @@
 
 from raiden.api.python import RaidenAPI
 from raiden.tests.utils.geth import wait_until_block
-from raiden.transfer.state import CHANNEL_STATE_SETTLED
+from raiden.transfer import views
+from raiden.transfer.state import NODE_NETWORK_REACHABLE
+from raiden import routing
 
 log = structlog.get_logger(__name__)
 
 
+def wait_for_reachable_node(sender, receiver):
+    """Wait until a node is reachable"""
+    while True:
+        state = RaidenAPI(sender).get_node_network_state(
+            receiver.address,
+        )
+        if state == NODE_NETWORK_REACHABLE:
+            return
+        gevent.sleep(1)
+
+
+def wait_for_transaction(
+    receiver,
+    registry_address,
+    token_address,
+    sender_address,
+):
+    """Wait until a first transaction in a channel is received"""
+    while True:
+        receiver_channel = RaidenAPI(receiver).get_channel_list(
+            registry_address=registry_address,
+            token_address=token_address,
+            partner_address=sender_address,
+        )
+        if(
```

Comment:
nitpick:

It would be a bit nicer to read if you put the condition in a variable and then did the `if` check on the variable. We do that a lot in the code for readability. e.g.:

```python
transaction_received = (
            len(receiver_channel) == 1 and
            receiver_channel[0].partner_state.balance_proof is not None
)
if transaction_received:
    break